lib/src/store.rs

```diff
@@ -81,11 +81,24 @@ impl Store {
         write_locked_cache.insert(id.clone(), data.clone());
         Ok(data)
     }
-
     pub fn write_commit(self: &Arc<Self>, commit: backend::Commit) -> BackendResult<Commit> {
         assert!(!commit.parents.is_empty());
-        let commit_id = self.backend.write_commit(&commit)?;
-        let data = Arc::new(commit);
+        let mut mut_commit = commit;
+        let commit_id = loop {
+            // It's possible that another commit with the same commit id (but different
+            // change id) already exists. Adjust the timestamp until this is no
+            // longer the case.
+            let maybe_commit_id = self.backend.write_commit(&mut_commit);
```

Comment:
Can we move the retry into the Git backend? Then we shouldn't have to add the `AlreadyHaveCommitWithId` variant. The error is specific to the Git backend, so it would be nice to not have it pollute the API.

Reply:
I tried that, and it turned into a mess. The git backend locks the repo for writing, and doing it naively makes it try to lock an already-locked `mut_repo` and hang forever. 

If I had to do it, I'd rename GitBackend::write_commit into `write_commit_internal` and wrap it into another function. Does that sound good to you?

Reply:
Sounds good. Thanks